src/client/action/watch.rs

```diff
@@ -0,0 +1,350 @@
+    /// collections.
+    ///
+    /// See the documentation [here](https://www.mongodb.com/docs/manual/changeStreams/) on change
+    /// streams.
+    ///
+    /// Change streams require either a "majority" read concern or no read concern. Anything else
+    /// will cause a server error.
+    pub fn watch(&self) -> Watch {
+        self.async_collection.watch()
+    }
+}
+
+mod private {
+    pub trait Sealed {}
+    impl Sealed for super::Implicit {}
+    impl<'a> Sealed for super::Explicit<'a> {}
+}
+
+pub trait Session: private::Sealed {}
+impl Session for Implicit {}
+impl<'a> Session for Explicit<'a> {}
+
+pub struct Implicit;
+pub struct Explicit<'a>(&'a mut ClientSession);
+
+/// Starts a new [`ChangeStream`] that receives events for all changes in the cluster.
+#[must_use]
+pub struct Watch<'a, S: Session = Implicit> {
+    client: &'a Client,
+    target: AggregateTarget,
+    pipeline: Vec<Document>,
+    options: Option<ChangeStreamOptions>,
+    session: S,
+}
+
+impl<'a> Watch<'a, Implicit> {
+    fn new(client: &'a Client, target: AggregateTarget) -> Self {
+        Self {
+            client,
+            target,
+            pipeline: vec![],
+            options: None,
+            session: Implicit,
+        }
+    }
+
+    fn new_cluster(client: &'a Client) -> Self {
+        let mut options = ChangeStreamOptions::default();
+        options.all_changes_for_cluster = Some(true);
+        Self {
+            client,
+            target: AggregateTarget::Database("admin".to_string()),
+            pipeline: vec![],
+            options: Some(options),
+            session: Implicit,
+        }
+    }
+}
+
+impl<'a, S: Session> Watch<'a, S> {
+    /// Apply an aggregation pipeline to the change stream.
+    ///
+    /// Note that using a `$project` stage to remove any of the `_id`, `operationType` or `ns`
+    /// fields will cause an error. The driver requires these fields to support resumability. For
+    /// more information on resumability, see the documentation for
+    /// [`ChangeStream`](change_stream/struct.ChangeStream.html)
+    ///
+    /// If the pipeline alters the structure of the returned events, the parsed type will need to be
+    /// changed via [`ChangeStream::with_type`].
+    pub fn pipeline(mut self, value: impl IntoIterator<Item = Document>) -> Self {
+        self.pipeline = value.into_iter().collect();
+        self
+    }
+
+    /// Specifies the logical starting point for the new change stream. Note that if a watched
+    /// collection is dropped and recreated or newly renamed, `start_after` should be set instead.
+    /// `resume_after` and `start_after` cannot be set simultaneously.
+    ///
+    /// For more information on resuming a change stream see the documentation [here](https://www.mongodb.com/docs/manual/changeStreams/#change-stream-resume-after)
+    pub fn resume_after(mut self, value: impl Into<Option<ResumeToken>>) -> Self {
```

Comment:
This was one scenario where it's clearly better to accept an `impl Into<Option<..>>` - if more of these show up I'm going to lean more towards having that be the standard setter parameter.

Reply:
Oh, sorry - because the user will get this value from `ChangeStream::resume_token`, which returns an `Option` ([example](https://github.com/abr-egn/mongo-rust-driver/blob/RUST-1512/better-helpers/src/test/documentation_examples.rs#L1813)).